qcodes/instrument_drivers/Keysight/keysightb1500/KeysightB1517A.py

```diff
@@ -287,3 +708,131 @@ def use_high_resolution_adc(self) -> None:
+                between v_start and v_end)
+            post_sweep_voltage_val: voltage to hold at end of sweep (i.e.
+                start or end val). Sweep chan will also output this voltage
+                if an abort condition is encountered during the sweep
+            av_coef: coefficient to use for av command to set ADC
+                averaging.  Negative value implies NPLC mode with absolute
+                value of av_coeff the NPLC setting to use. Positive value
+                implies auto mode and must be set to >= 4
+            enable_filter: turn SMU filter on or off
+            v_src_range: range setting to use for voltage source
+            i_comp: current compliance level
+            i_meas_range: current measurement range
+            hold_time: time (in s) to wait before starting very first
+                measurement in sweep
+            delay: time (in s) after starting to force a step output and
+                before starting a step measurement
+            step_delay: time (in s) after starting a step measurement before
+                next step in staircase. If step_delay is < measurement time,
+                B1500 waits until measurement complete and then forces the
+                next step value.
+            measure_delay: time (in s)  after receiving a start step
+                measurement trigger and before starting a step measurement
+            abort_enabled: Enbale abort
+            sweep_mode: Linear, log, linear-2-way or log-2-way
+          """
+        self.set_average_samples_for_high_speed_adc(av_coef)
+        self.enable_filter(enable_filter)
+        self.source_config(output_range=v_src_range,
+                           compliance=i_comp,
+                           min_compliance_range=i_meas_range)
+        self.voltage(v_start)
+        self.measurement_operation_mode(constants.CMM.Mode.COMPLIANCE_SIDE)
+        self.current_measurement_range(i_meas_range)
+        self.iv_sweep.hold_time(hold_time)
+        self.iv_sweep.delay(delay)
+        self.iv_sweep.step_delay(step_delay)
+        self.iv_sweep.measure_delay(measure_delay)
+        self.iv_sweep.sweep_auto_abort(abort_enabled)
+        self.iv_sweep.post_sweep_voltage_condition(post_sweep_voltage_val)
+        self.iv_sweep.sweep_mode(sweep_mode)
+        self.iv_sweep.sweep_range(v_src_range)
+        self.iv_sweep.sweep_start(v_start)
+        self.iv_sweep.sweep_end(v_end)
+        self.iv_sweep.sweep_steps(n_steps)
+        self.iv_sweep.current_compliance(i_comp)
+        self.root_instrument.clear_timer_count()
+
+        error_list, error = [], ''
+
+        while error != '+0,"No Error."':
+            error = self.root_instrument.error_message()
+            error_list.append(error)
+
+        if len(error_list) <= 1:
+            self.setup_fnc_already_run = True
+        else:
+            raise Exception(f'Received following errors while trying to set '
```

Comment:
Suggest making this a RuntimeError or some other error that is a bit more specific

Reply:
Thanks. Made this a runtume error.